src/core/Akka/Actor/Futures.cs

```diff
@@ -176,16 +176,22 @@ public static Task<T> Ask<T>(this ICanTell self, Func<IActorRef, object> message
         /// <returns>Provider used for Ask pattern implementation</returns>
         internal static IActorRefProvider ResolveProvider(ICanTell self)
         {
-            if (self is ActorSelection selection)
-                return ResolveProvider(selection.Anchor);
-
-            if (self is IInternalActorRef actorRef)
-                return actorRef.Provider;
+            while (true)
+            {
+                switch (self)
+                {
+                    case ActorSelection selection:
+                        self = selection.Anchor;
```

Comment:
Crazy question: What would the cost/benefit be of having a 'hardanchor' passed around?
Even if the prop/method itself is a reverse-recursive (playing with stack fire?) it may help if there's other cases that need this.

Reply:
What's a "hardanchor" in this case?

Reply:
all I did here, btw, was just unroll the tail recursion using Rider.